### ISSUE TITLE
Can't Run Unity Tests via Unity Editor or Standalone NUnit Launcher

### ISSUE THREAD

**Issue by wiverson** [JetBrains/resharper-unity#1247]
Created a new empty project with Unity 2019.1.10f1 on Mac OS X 10.14.5.  Using Rider 2019.1.3.

JetBrains Rider 2019.1.3
Build #RD-191.7141.460, built on July 9, 2019
JRE: 1.8.0_212-release-1586-b4 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.5

In Unity, created a Test Assembly Folder, and then added a test.  When I hit run on the tests in Unity itself, tests run fine.  When I switch to Rider and hit with the tests set to run in the Unity Editor, tests all report as Inconclusive.

Would really like to have non-Unity dependent code run in the Standalone NUnit Launcher.

Here's the log file output for Unity Editor - Edit Mode:

```
15:57:16.436 |T| Launch: 7198e9d3-6932-4e05-bed1-eca0b9d737ad >> RunHostController.AfterLaunchStarted
15:57:16.436 |T| Launch: 7198e9d3-6932-4e05-bed1-eca0b9d737ad << RunHostController.AfterLaunchStarted
15:57:16.437 |I| BuildStage started
15:57:16.437 |I| Build policy: Automatic
15:57:16.437 |I| Skipping build of open folder projects
15:57:16.437 |I| BuildStage completed
15:57:16.437 |I| RefreshProjectPropertiesStage started
15:57:16.437 |I| RefreshProjectPropertiesStage completed
15:57:16.437 |I| DiscoveryStage started
15:57:16.438 |I| Test exploration required for 0 projects
15:57:16.438 |I| DiscoveryStage completed
15:57:16.438 |I| BuildPipelineStage started
15:57:16.438 |I| BuildPipelineStage completed
15:57:16.438 |I| ApplyRulesStage started
15:57:16.439 |V| Running elements from criterion: {ConjunctiveCriterion: {DisjunctiveCriterion: {TestAncestorCriterion: NUnit3x::E5A67615-3685-D42A-6C0B-47A16E75AAED::.NETFramework,Version=v4.7.1::Tests.FancyUnityTestScript}, {TestAncestorCriterion: NUnit3x::E5A67615-3685-D42A-6C0B-47A16E75AAED::.NETFramework,Version=v4.7.1::Tests.NunitTestClass.TestSomething}}, {TestAncestorCriterion: NUnit3x::E5A67615-3685-D42A-6C0B-47A16E75AAED::.NETFramework,Version=v4.7.1::Tests.NunitTestClass}}
15:57:16.439 |I| Got 1 elements (1 explicit) to run
15:57:16.440 |V| Marking 1 elements as Pending
15:57:16.440 |V| Marking elements as Pending completed
15:57:16.440 |V| Distributing 1 elements into runs
15:57:16.440 |V| Distributing elements completed
15:57:16.440 |I| Created 1 unit-test runs
15:57:16.440 |I| ApplyRulesStage completed
15:57:16.440 |I| RunTestsStage started
15:57:16.440 |I| Running up to 1 unit-test runs in parallel
15:57:16.441 |I| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 - Starting
15:57:16.441 |V|     Provider: NUnit 3x
    Target Framework: .NETFramework,Version=v4.7.1
    Strategy: RunViaUnityEditorStrategy
    Runtime Enviroment: JetBrains.ReSharper.Plugins.Unity.Rider.UnitTesting.RunViaUnityEditorStrategy+UnityRuntimeEnvironment
15:57:16.441 |T| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 >> RunHostController.PrepareForRun
15:57:16.441 |T| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 << RunHostController.PrepareForRun
15:57:17.905 |T| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 >> RunHostController.CleanupAfterRun
15:57:17.905 |T| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 << RunHostController.CleanupAfterRun
15:57:17.905 |I| Run: 019a7fa5-9e5a-47ff-9440-ef328d96fdf9 - Finished
15:57:17.905 |I| RunTestsStage completed
15:57:17.905 |I| FinishLaunchStage started
15:57:17.906 |I| Finishing launch
15:57:17.906 |W| Element Tests.NunitTestClass.TestSomething was left pending after its run completion.
15:57:17.907 |I| Launch is finished
15:57:17.907 |I| FinishLaunchStage completed
15:57:17.907 |T| Launch: 7198e9d3-6932-4e05-bed1-eca0b9d737ad >> RunHostController.BeforeLaunchFinished
15:57:17.907 |T| Launch: 7198e9d3-6932-4e05-bed1-eca0b9d737ad << RunHostController.BeforeLaunchFinished

```

When I switch to the Standalone NUnit Launcher, I get a different error message:

```
15:59:21.554 |T| Launch: 90a4fd3b-a65b-4ed1-85cd-88545e90e96e >> RunHostController.AfterLaunchStarted
15:59:21.554 |T| Launch: 90a4fd3b-a65b-4ed1-85cd-88545e90e96e << RunHostController.AfterLaunchStarted
15:59:21.554 |I| BuildStage started
15:59:21.555 |I| Build policy: Automatic
15:59:22.627 |I| 0 projects built
15:59:22.627 |I| BuildStage completed
15:59:22.628 |I| RefreshProjectPropertiesStage started
15:59:22.628 |I| RefreshProjectPropertiesStage completed
15:59:22.628 |I| DiscoveryStage started
15:59:22.629 |I| Test exploration required for 0 projects
15:59:22.629 |I| DiscoveryStage completed
15:59:22.629 |I| BuildPipelineStage started
15:59:22.629 |I| BuildPipelineStage completed
15:59:22.629 |I| ApplyRulesStage started
15:59:22.630 |V| Running elements from criterion: {ConjunctiveCriterion: {DisjunctiveCriterion: {TestAncestorCriterion: NUnit3x::E5A67615-3685-D42A-6C0B-47A16E75AAED::.NETFramework,Version=v4.7.1::Tests.FancyUnityTestScript}, {TestAncestorCriterion: NUnit3x::E5A67615-3685-D42A-6C0B-47A16E75AAED::.NETFramework,Version=v4.7.1::Tests.NunitTestClass.TestSomething}}, {ProjectCriterion: /Users/wiverson/Desktop/nunit-test/NUnitTest}}
15:59:22.630 |I| Got 3 elements (0 explicit) to run
15:59:22.630 |V| Marking 3 elements as Pending
15:59:22.631 |V| Marking elements as Pending completed
15:59:22.631 |V| Distributing 3 elements into runs
15:59:22.632 |V| Distributing elements completed
15:59:22.632 |I| Created 1 unit-test runs
15:59:22.632 |I| ApplyRulesStage completed
15:59:22.632 |I| RunTestsStage started
15:59:22.632 |I| Running up to 1 unit-test runs in parallel
15:59:22.632 |I| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd - Starting
15:59:22.632 |V|     Provider: NUnit 3x
    Target Framework: .NETFramework,Version=v4.7.1
    Strategy: NUnitOutOfProcessUnitTestRunStrategy
    Runtime Enviroment: TaskRunnerRuntimeEnvironment
  TargetPlatform: MSIL
  TargetFramework: .NETFramework,Version=v4.7.1
  IsUnmanaged: False
  PlatformMonoPreference: PreferClr
15:59:22.632 |T| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd >> RunHostController.PrepareForRun
15:59:22.632 |T| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd << RunHostController.PrepareForRun
15:59:22.633 |I| [ProcessStartInfoPatchResult]
  Original request: FrameworkRequest (UseInternal: False, UseMono: False, MonoDebugMode: True, MonoPlatform: Automatic)
  Success: True
  ErrorMessage: <Empty>
  Original StartInfo: FileName = '/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/ReSharperTestRunnerIL.exe', Arguments = '127.0.0.1 57873 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd /Launcher::AssemblyFile:"/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/JetBrains.ReSharper.TaskRunnerFramework.dll" /Launcher::ClassName:JetBrains.ReSharper.TaskRunnerFramework.TaskExecutor /Launcher::MethodName:RunCommandLine /Launcher::Mode:InvokeMethod /Launcher::NonInteractive /Launcher::NoSplash /Launcher::AppDomainBaseDir:/Users/wiverson/Library/Caches/Rider2019.1/resharper-host/temp/UnitTestRunnerFakeBaseDir /Launcher::TargetFrameworkName:.NETFramework,Version=v4.7.1', WorkDir = '/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost'
  Patched  StartInfo: FileName = '/Library/Frameworks/Mono.framework/Versions/Current/bin/mono', Arguments = '--debug "/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/ReSharperTestRunnerIL.exe" 127.0.0.1 57873 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd /Launcher::AssemblyFile:"/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/JetBrains.ReSharper.TaskRunnerFramework.dll" /Launcher::ClassName:JetBrains.ReSharper.TaskRunnerFramework.TaskExecutor /Launcher::MethodName:RunCommandLine /Launcher::Mode:InvokeMethod /Launcher::NonInteractive /Launcher::NoSplash /Launcher::AppDomainBaseDir:/Users/wiverson/Library/Caches/Rider2019.1/resharper-host/temp/UnitTestRunnerFakeBaseDir /Launcher::TargetFrameworkName:.NETFramework,Version=v4.7.1', WorkDir = '/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost'

15:59:22.634 |I| Starting process: /Library/Frameworks/Mono.framework/Versions/Current/bin/mono --debug "/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/ReSharperTestRunnerIL.exe" 127.0.0.1 57873 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd /Launcher::AssemblyFile:"/Users/wiverson/Library/Application Support/JetBrains/Toolbox/apps/Rider/ch-0/191.7141.460/Rider.app/Contents/lib/ReSharperHost/JetBrains.ReSharper.TaskRunnerFramework.dll" /Launcher::ClassName:JetBrains.ReSharper.TaskRunnerFramework.TaskExecutor /Launcher::MethodName:RunCommandLine /Launcher::Mode:InvokeMethod /Launcher::NonInteractive /Launcher::NoSplash /Launcher::AppDomainBaseDir:/Users/wiverson/Library/Caches/Rider2019.1/resharper-host/temp/UnitTestRunnerFakeBaseDir /Launcher::TargetFrameworkName:.NETFramework,Version=v4.7.1
15:59:22.658 |I| Started process with pid 52018
15:59:23.292 |E| System.TypeLoadException: Could not resolve type with token 0100001d from typeref (expected class 'System.Xml.Linq.XElement' in assembly 'System.Xml.Linq, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089')
System.TypeLoadException: Could not resolve type with token 0100001d from typeref (expected class 'System.Xml.Linq.XElement' in assembly 'System.Xml.Linq, Version=3.5.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089')
  at JetBrains.ReSharper.UnitTestRunner.nUnit30.BuiltInNUnitRunner+<>c__DisplayClass1_0.<RunTests>b__0 () [0x00039] in C:\BuildAgent\work\49bcdd6b9dae3ff1\Psi.Features\UnitTesting\nUnit\Runner30\Src\BuiltInNUnitRunner.cs:37 
  at JetBrains.ReSharper.UnitTestRunner.nUnit30.BuiltInNUnitRunner.WithExtensiveErrorHandling (JetBrains.ReSharper.TaskRunnerFramework.IRemoteTaskServer server, System.Action action) [0x00000] in C:\BuildAgent\work\49bcdd6b9dae3ff1\Psi.Features\UnitTesting\nUnit\Runner30\Src\BuiltInNUnitRunner.cs:73 

15:59:23.309 |I| Process /Library/Frameworks/Mono.framework/Versions/Current/bin/mono:52018 has exited with code (0)
Output stream: 
15:59:23.309 |V| Trying to removing cache folder /var/folders/7n/v29swgs16kj3v_zfv3qg419m0000gn/T/7h8dzlds.e9j
15:59:23.310 |T| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd >> RunHostController.CleanupAfterRun
15:59:23.310 |T| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd << RunHostController.CleanupAfterRun
15:59:23.310 |I| Run: 86b94ae4-3a5f-4aaf-972c-cd3fe5ccd1cd - Finished
15:59:23.310 |I| RunTestsStage completed
15:59:23.310 |I| FinishLaunchStage started
15:59:23.311 |I| Finishing launch
15:59:23.311 |W| Element Tests.FancyUnityTestScript.FancyUnityTestScriptSimplePasses was left pending after its run completion.
15:59:23.311 |W| Element Tests.NunitTestClass.TestSomething was left pending after its run completion.
15:59:23.311 |I| Launch is finished
15:59:23.311 |I| FinishLaunchStage completed
15:59:23.311 |T| Launch: 90a4fd3b-a65b-4ed1-85cd-88545e90e96e >> RunHostController.BeforeLaunchFinished
15:59:23.311 |T| Launch: 90a4fd3b-a65b-4ed1-85cd-88545e90e96e << RunHostController.BeforeLaunchFinished

```

**Comment by wiverson** [comment]
Here is the super-fancy NUnit test case I'd like to run in the Standalone NUnit Launcher, lol.

```
using NUnit.Framework;

namespace Tests
{
    [TestFixture]
    public class NunitTestClass
    {
        [Test]
        public void TestSomething()
        {
            Assert.IsTrue(true);
        }
    }
}
```

**Comment by wiverson** [comment]
tl;dr - tests have to be in an Edit Mode test folder - Play Mode won't work.

So, I created an Edit Mode test folder and moved my tests to that folder, and now everything is working fine.

Closing, hopefully this might be helpful for the next person trying to Google for this.  ;)
